ui/app/components/Breadcumb.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import type { RouteLocationNormalizedLoaded } from "vue-router";
+
+const route: RouteLocationNormalizedLoaded = useRoute();
+
+const props = withDefaults(
+  defineProps<{
+    showHomeIcon: boolean;
+    customLabels: CustomLabels;
+  }>(),
+  {
+    showHomeIcon: true,
+    customLabels: () => ({}),
+  }
+);
+
+// Function to capitalize and format path segments
+const formatPathSegment = (segment: string): string => {
+  // Check if there's a custom label first
+  if (props.customLabels[segment]) {
+    return props.customLabels[segment];
+  }
+
+  // Remove hyphens and underscores, then capitalize
+  return segment
+    .split(/[-_]/)
+    .map((word: string) => word.charAt(0).toUpperCase() + word.slice(1))
```

Comment:
See if something like `https://unjs.io/packages/scule#titlecasestr-opts-normalize` works for you, if not leave this function